engine/configurator/conf_cyaml.c

```diff
@@ -0,0 +1,258 @@
+    {
+        /**< Success. */
+        case CYAML_OK:
+            return 0;
+
+        /* Memory allocation failed. */
+        case CYAML_ERR_OOM:
+            ERROR("There is a CYAML error %d", err);
+            return TE_ENOMEM;
+
+        /* See \ref CYAML_CFG_NO_ALIAS. */
+        case CYAML_ERR_ALIAS:
+        /* Failed to open file. */
+        case CYAML_ERR_FILE_OPEN:
+        /* Mapping key rejected by schema. */
+        case CYAML_ERR_INVALID_KEY:
+        /* Value rejected by schema. */
+        case CYAML_ERR_INVALID_VALUE:
+        /* No anchor found for alias. */
+        case CYAML_ERR_INVALID_ALIAS:
+        /* Internal error in LibCYAML. */
+        case CYAML_ERR_INTERNAL_ERROR:
+        /* YAML event rejected by schema. */
+        case CYAML_ERR_UNEXPECTED_EVENT:
+        /* String length too short. */
+        case CYAML_ERR_STRING_LENGTH_MIN:
+        /* String length too long. */
+        case CYAML_ERR_STRING_LENGTH_MAX:
+        /* Value's data size unsupported. */
+        case CYAML_ERR_INVALID_DATA_SIZE:
+        /* Top level type must be pointer. */
+        case CYAML_ERR_TOP_LEVEL_NON_PTR:
+        /* Schema contains invalid type. */
+        case CYAML_ERR_BAD_TYPE_IN_SCHEMA:
+        /* Schema minimum exceeds maximum. */
+        case CYAML_ERR_BAD_MIN_MAX_SCHEMA:
+        /* Bad seq_count param for schema. */
+        case CYAML_ERR_BAD_PARAM_SEQ_COUNT:
+        /* Client gave NULL data argument. */
+        case CYAML_ERR_BAD_PARAM_NULL_DATA:
+        /* Bit value beyond bit field size. */
+        case CYAML_ERR_BAD_BITVAL_IN_SCHEMA:
+        /* Too few sequence entries. */
+        case CYAML_ERR_SEQUENCE_ENTRIES_MIN:
+        /* Too many sequence entries. */
+        case CYAML_ERR_SEQUENCE_ENTRIES_MAX:
+        /* Mismatch between min and max. */
+        case CYAML_ERR_SEQUENCE_FIXED_COUNT:
+        /* Non-fixed sequence in sequence. */
+        case CYAML_ERR_SEQUENCE_IN_SEQUENCE:
+        /* Required mapping field missing. */
+        case CYAML_ERR_MAPPING_FIELD_MISSING:
+        /* Client gave NULL mem function. */
+        case CYAML_ERR_BAD_CONFIG_NULL_MEMFN:
+        /* Client gave NULL config arg. */
+        case CYAML_ERR_BAD_PARAM_NULL_CONFIG:
+        /* Client gave NULL schema arg. */
+        case CYAML_ERR_BAD_PARAM_NULL_SCHEMA:
+        /* Failed to initialise libyaml. */
+        case CYAML_ERR_LIBYAML_EMITTER_INIT:
+        /* Failed to initialise libyaml. */
+        case CYAML_ERR_LIBYAML_PARSER_INIT:
+        /* Failed to initialise libyaml. */
+        case CYAML_ERR_LIBYAML_EVENT_INIT:
+        /* Error inside libyaml emitter. */
+        case CYAML_ERR_LIBYAML_EMITTER:
+        /* Error inside libyaml parser. */
+        case CYAML_ERR_LIBYAML_PARSER:
+        /* Count of CYAML return codes. */
+        case CYAML_ERR__COUNT:
+        {
+            ERROR("There is a CYAML error %s(%d)",
+                  cyaml_strerror(err), err);
+            return TE_EINVAL;
+        }
```

Comment:
The point was that maybe one should process CYAML errors correctly. Not just by sending TE_EINVAL. But I'm not sure that I understand how to process these errors correctly and do not ignore them. It's easy to correct the written code or remove it.

Reply:
The problem is not why the function exists, but how it is written. It's not a good idea to list 3rd party error codes and then do nothing about them, because it's impossible to keep things in sync. You should convert the values you know how to convert and then just use `default` for everything else.

Reply:
I thought that may be someone could know how process these errors. And maybe I'm wrong that there is always TE_EINVAL. OK, I'll hide in `default`.